File: LearnCSharpTests/ProgramTests.cs
using System;
using System.IO;
using LearnCSharpLibrary;
using Xunit;

namespace LearnCSharpTests
{
    public class ProgramTests
    {
        [Fact]
        public void MainTest()
        {
            var consoleOutput = new StringWriter();
            Console.SetOut(consoleOutput);

            var bootstrapTest = new Bootstrap(new []{"MainTest"});
            bootstrapTest.Start();

            Assert.Equal("Hello MainTest!\n", consoleOutput.ToString());
        }
    }
}

File: LearnVBTests/Console/ProgramTests.vb
Imports System.IO
Imports LearnVBLibrary
Imports Xunit

Public Class ProgramTests
    <Fact>
    Sub MainTest()
        Dim consoleOutput = New StringWriter
        Console.SetOut(consoleOutput)

        Dim bootstrapTest = New Bootstrap({"MainTest"})
        bootstrapTest.Start()

        Assert.Equal($"Hello World from MainTest!{Environment.NewLine}", consoleOutput.ToString())
    End Sub
End Class
